deployment/terraform/api.tf

```diff
@@ -0,0 +1,143 @@
+#
+# Security Group Resources
+#
+resource "aws_security_group" "alb" {
+  name   = "sg${var.project}APILoadBalancer"
+  vpc_id = data.terraform_remote_state.core.outputs.vpc_id
+
+  tags = {
+    Name    = "sg${var.project}APILoadBalancer",
+    Project = var.project
+  }
+}
+
+#
+# ALB Resources
+#
+resource "aws_lb" "api" {
+  name            = "alb${var.project}API"
+  security_groups = [aws_security_group.alb.id]
+  subnets         = data.terraform_remote_state.core.outputs.public_subnet_ids
+
+  enable_http2 = true
+
+  tags = {
+    Name    = "alb${var.project}API"
+    Project = var.project
+  }
+}
+
+resource "aws_lb_target_group" "api" {
+  name = "tg${var.project}API"
+
+  health_check {
+    healthy_threshold   = 3
+    interval            = 30
+    matcher             = 200
+    protocol            = "HTTP"
+    timeout             = 3
+    path                = "/api/hello/world"
+    unhealthy_threshold = 2
+  }
+
+  port     = 80
+  protocol = "HTTP"
+  vpc_id   = data.terraform_remote_state.core.outputs.vpc_id
+
+  tags = {
+    Name    = "tg${var.project}API"
+    Project = var.project
+  }
+}
+
+resource "aws_lb_listener" "api_redirect" {
+  load_balancer_arn = aws_lb.api.arn
+  port              = 80
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = 443
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "api" {
+  load_balancer_arn = aws_lb.api.id
+  port              = 443
+  protocol          = "HTTPS"
+  certificate_arn   = data.terraform_remote_state.core.outputs.ssl_certificate_arn
```

Comment:
This one may require a more complex solution. We should have the ability to provision our own certificate in alternate environments, and I think it would be good for us to take advantage of that. We may need to create an individual ACM certificate for just the Granary subdomain so that we can have the ability to manage future domains in a similar way.